elastic-agent-client.proto

```diff
@@ -101,26 +129,311 @@ message ActionResponse {
         // Action has failed.
         FAILED = 1;
     }
-    // Token that is used to unique identify the application to agent. When agent started this
+    // Token that is used to uniquely identify the application to agent. When agent started this
     // application it would have provided it this token.
     string token = 1;
     // Unique ID of the action.
     string id = 2;
     // Status of the action.
     Status status = 3;
-    // JSON encoded result for the action.
+    // JSON encoded result for the action (empty when diagnostic action response).
     bytes result = 4;
+    // Specific result for diagnostics action. (only used in V2)
+    repeated ActionDiagnosticUnitResult diagnostic = 5;
+}
+
+// State codes for the current state.
+enum State {
+    STARTING = 0;
+    CONFIGURING = 1;
+    HEALTHY = 2;
+    DEGRADED = 3;
+    FAILED = 4;
+    STOPPING = 5;
+}
+
+// Type of unit.
+enum UnitType {
+    INPUT = 0;
+    OUTPUT = 1;
+}
+
+// A unit that is part of a collector/shipper.
```

Comment:
There is no requirement saying that it can only have 1. In the future might be possible to have a shipper that sends the same event to 2 different outputs.